testhelper/integrationtesthelper.go

```diff
@@ -533,23 +662,42 @@ func mockLidarReadingsValid() error {
 	return nil
 }
 
-func mockIMUReadingsValid(t *testing.T) ([]string, error) {
-	file, err := os.Open(artifact.MustPath(mockDataPath + "/imu/data.txt"))
+func mockMovementSensorReadingsValid(t *testing.T) (movementSensorData, error) {
+	file, err := os.Open(artifact.MustPath(mockDataPath + "/movement_sensor/data.json"))
 	if err != nil {
-		t.Error("TEST FAILED TimedIMUSensorReading Mock failed to open data file")
-		return []string{}, err
+		t.Error("TEST FAILED TimedMovementSensorReading Mock failed to open data file")
+		return movementSensorData{}, err
 	}
+	defer func() {
+		if err := file.Close(); err != nil {
+			t.Error("TEST FAILED TimedMovementSensorReading Mock failed to close data file")
+		}
+		test.That(t, err, test.ShouldBeNil)
+	}()
 
-	mockDatasetScanner := bufio.NewScanner(file)
-	mockDatasetScanner.Split(bufio.ScanLines)
-	var mockDataset []string
+	byteValue, err := io.ReadAll(file)
+	if err != nil {
+		t.Error("TEST FAILED TimedMovementSensorReading Mock failed to read data file")
+		return movementSensorData{}, err
+	}
+
+	var data movementSensorData
+
+	if err = json.Unmarshal(byteValue, &data); err != nil {
+		t.Error("TEST FAILED TimedMovementSensorReading Mock failed to unmarshal json")
+		return movementSensorData{}, err
+	}
 
-	for mockDatasetScanner.Scan() {
-		mockDataset = append(mockDataset, mockDatasetScanner.Text())
+	if len(data.AngVelData) != len(data.LinAccData) &&
```

Comment:
[nit] 
```
expectedDataNum := len(data.AngVelData)
if expectedDataNum!= len(data.LinAccData) &&
		expectedDataNum != len(data.OrientationData) &&
		expectedDataNum != len(data.PosData) {
....

if expectedDataNum < NumMovementSensorData {
```

Also we may want to check that the `expectedDataNum >NumMovementSensorData` before comparing to all other data fields